skins/laika/src/components/CookieNotice.vue

```diff
@@ -0,0 +1,143 @@
+									:content="$t('cookie_option_required_content')"
+									:checked="true"
+									:disabled="true"
+								></CookieCheckbox>
+							</li>
+							<li>
+								<CookieCheckbox
+									:heading="$t('cookie_option_optional_heading')"
+									:content="$t('cookie_option_optional_content')"
+									:checked="optionalChecked"
+									:disabled="false"
+									v-on:toggle="onOptionalToggle"
+								></CookieCheckbox>
+							</li>
+						</ul>
+					</div>
+				</form>
+			</div>
+			<div class="cookie-notice-buttons">
+				<div class="cookie-notice-button check">
+					<button class="button is-primary is-main is-outlined" v-on:click="onCheckButtonClick">
+						<span>{{ $t( 'cookie_button_check' ) }}</span>
+					</button>
+				</div>
+				<div class="cookie-notice-button accept">
+					<button class="button is-primary is-main is-outlined" v-on:click="removeNotice">
+						<span>{{ $t( 'cookie_button_accept' ) }}</span>
+					</button>
+				</div>
+			</div>
+		</div>
+		<div :style="{ height: height + 'px' }"></div>
+	</div>
+</template>
+
+<script>
+import Vue from 'vue';
+import { ref, onMounted, onUnmounted } from '@vue/composition-api';
+import CookieCheckbox from './CookieCheckbox';
+
+export default Vue.extend( {
+	name: 'CookieNotice',
+	components: {
+		CookieCheckbox,
+	},
+	props: {
+		showCookieNotice: Boolean,
+	},
+	setup( props ) {
+		const isVisible = ref( props.showCookieNotice );
+		const cookieNotice = ref( null );
+		const height = ref( 0 );
+		const optionalChecked = ref( false );
+		const showOptions = ref( false );
+		const textOpen = ref( false );
+
+		const toggleTextOpen = ( e ) => {
+			e.preventDefault();
+			textOpen.value = !textOpen.value;
+		};
+
+		const onContentResize = () => {
+			height.value = cookieNotice.value.offsetHeight;
+		};
+
+		const removeNotice = () => {
+			height.value = 0;
+			isVisible.value = false;
+			cookieNotice.value.removeEventListener( 'click', onContentResize );
+			window.removeEventListener( 'resize', onContentResize );
+		};
+
+		const onOptionalToggle = () => {
+			optionalChecked.value = !optionalChecked.value;
+		};
+
+		const onCheckButtonClick = ( e ) => {
+			e.preventDefault();
+			if ( !showOptions.value ) {
+				showOptions.value = true;
+			} else {
+				removeNotice();
+			}
+		};
+
+		const onBackButtonClick = ( e ) => {
+			e.preventDefault();
+			showOptions.value = false;
+		};
+
+		onMounted( () => {
+			if ( !isVisible.value ) {
+				return;
+			}
+			height.value = cookieNotice.value.offsetHeight;
+			cookieNotice.value.addEventListener( 'click', onContentResize );
+			window.addEventListener( 'resize', onContentResize );
+		} );
+
+		onUnmounted( () => {
+			removeNotice();
+		} );
+
+		return {
+			isVisible,
+			cookieNotice,
+			height,
+			optionalChecked,
+			showOptions,
+			textOpen,
+			toggleTextOpen,
+			onOptionalToggle,
+			onCheckButtonClick,
+			onBackButtonClick,
+			removeNotice,
+		};
+	},
+} );
+</script>
```

Comment:
This component is doing three things at once, making it a bit hard to follow:

- Act as a container, with height adjustment
- A paging mechanism (page 1 and 2, with the ability to go back and forth)
- Keeping the state of the optional buttons.

Maybe you can split this into several components, e.g. container, "paging" and 2 pages, maybe reusing the button components on both pages. This will make the template easier to read and distribute the state a bit.